### DISCUSSION TITLE
Templated object views

### DISCUSSION THREAD

**Discussion by johanfforsberg** [nasa/openmct#5334] [Community Q&A]
Hi,

I'm thinking of a use case for OpenMCT where there is a large tree of objects with several levels in a strict type hierarchy. Since there can be large numbers of objects of different kinds at each level, it would be a lot of work to make custom views for each of them, as they can have very different requirements. 

It would be really great if the end-users were able to customize such views themselves by using layouts that were "templated", instead of having to write code. What I mean is that, instead of hard-coded to show specific objects, the layout would contain relative "pointers" to things available under the object being viewed. The template view would not have to be very intelligent about it, as it would be built with knowledge of what's available.

Is something like this already possible? If not, I'd be interested in building a plugin for it, and would appreciate any clues.

**Comment by serbuh** [comment]
Check out this thread in discussions:
[Folders hierarchy #7494](https://github.com/nasa/openmct/discussions/7494)
